### PR TITLE
Discord plugin

### LINKED ISSUE #513: Discord Webhook Plugin
**Is your feature request related to a problem? Please describe.**
No, this is an additional feature. Due to the growing Tezos Discord communities, this would be a nice additional feature for many TRD users.

**Describe the solution you'd like**
I would like to send reward notifications to Discord. However, the current webhook plugin does not support the correct format.

**Describe alternatives you've considered**
Create a proxy function that takes the current TRD webhook data and transforms it into Discord compatible data and sends this to the Discord webhook. 

**Additional context**
I would like to have something like this:
image
and not every payout data the webhook currently submits.
Discord Webhook Resource: https://discord.com/developers/docs/resources/webhook

### PR DESCRIPTION
**Pull request by utdrmac**
Resolves #513

* **Performed tests**: Performed various tests on own discord server.
* **Check list**:

- [x] I extended the Github Actions CI test units with the corresponding tests for this new feature (if needed).
- [x] I extended the Sphinx documentation (if needed).
- [x] I extended the help section (if needed).
- [x] I changed the README file (if needed).
- [x] I created a new issue if there is further work left to be done (if needed).

**Work effort**: 4hrs

### CODE REVIEW

**Comment by jdsika** [comment]
@utdrmac if you make the checks green, Daniel will test the branch with our Discord server. Thank you for this!

**Comment by dansan566** [comment]
Many thanks for this great implementation. It works (tested in prod)! However, is there a way to suppress the admin messages? Furthermore, a Dry-Run does not send notifications? Is this the desired behavior?
Bildschirmfoto 2021-12-07 um 13 52 07

**Comment by utdrmac** [comment]
@dansan566 Correct, dry-run does not invoke any of the plugins.
I can remove the admin message. I wasn't sure what all to display.

**Comment by dansan566** [comment]
@utdrmac Can you implement a on/off option like this:
```
plugins:
  enabled:
    - discord
  discord:
    endpoint: https://discord.com/api/webhooks/9876543212345678/OmAfadfasdfasdfasdfasdfasdfasfsdf
    admin_message: off
    discord_text: >
      Rewards for cycle %CYCLE% are completed.
      We paid out %TREWARDS% XTZ in rewards to %NDELEGATORS% delegators.   
```

**Comment by utdrmac** [comment]
> Can you implement a on/off option like this:

Done

**Comment by dansan566** [comment]
So I need to add `admin_message: FALSE` or `admin_message: TRUE`?

**Comment by utdrmac** [comment]
> So I need to add `admin_message: FALSE` or `admin_message: TRUE`?

`send_admin: False`

**Comment by jdsika** [comment]
Can I configure duplicated plugins, one with admin message for one channel and the other without admin messages for another?

File: docs/plugins.md [hunk 1]
```diff
@@ -153,6 +153,35 @@ in the cycle and paid out 1234.98 XTZ in rewards.
...
+This plugin makes an HTTP POST request to a Discord Webhook URL.
+
+You must be an admin of a discord server to generate a webhook URL. Do so by going to 'Server Settings' -> 'Integrations' -> 'New Webhook'. Select the name, upload an avatar icon (if desired), and select which channel the messages will appear.
+
+Paste the generated URL into the appropriate field in the plugin section. You can customize the message for 'payouts' notifications. This is handled exactly as in the twitter plygin
+
+### Example Config
+
+* endpoint: The webhook URL generated by your discord server
+* send_admin: Whether or not to send administrative-style messages using the bot. These messages will appear alongside general user-style messages.
```
Comment (jdsika) [inline docs/plugins.md:167]: In general I would be a fan of defining the type like <bool> or <string> but maybe the example is enough... or false/true

File: src/plugins/discord.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            if k not in cfg_keys:
+                raise plugins.PluginConfigurationError("[{:s}] '{:s}' setting not found".format(self.name, k))
+
+        # Set config
+        self.endpoint = self.cfg["endpoint"]
+        self.discord_text = self.cfg["discord_text"]
+        self.send_admin = self.cfg["send_admin"]
+
+        # Sanity
+        if self.endpoint is None:
```
Comment (jdsika) [inline src/plugins/discord.py:69]: no sanity check like is_bool() ?

**Review by jdsika — CHANGES_REQUESTED** [review]

File: src/plugins/discord.py (hunk 2 above)
Comment (utdrmac) [reply src/plugins/discord.py:69]: ConfigParser already handles that. It parses 'send_admin' into a bool for you.

**Review by jdsika — APPROVED** [review]

**Pull request merged**